encrypt/encrypt.go

```diff
@@ -1,43 +1,350 @@
+		if _, err := ChecksumFileUnencMd5.WriteString(fmt.Sprintf("%s %s\n", hashes.unencryptedMd5, file.unencrypted)); err != nil {
+			return err
+		}
+
+		if _, err := ChecksumFileUnencSha256.WriteString(fmt.Sprintf("%s %s\n", hashes.unencryptedSha256, file.unencrypted)); err != nil {
+			return err
+		}
+
+		if _, err := ChecksumFileEncMd5.WriteString(fmt.Sprintf("%s %s\n", hashes.encryptedMd5, file.encrypted)); err != nil {
+			return err
+		}
+
+		if _, err := ChecksumFileEncSha256.WriteString(fmt.Sprintf("%s %s\n", hashes.encryptedSha256, file.encrypted)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Checks that all the input files exists, and are readable, and that the
+// output files do not exist
+func checkFiles(files []encryptionFileSet) error {
+	log.Info("Checking files")
+	for _, file := range files {
+		// check that the input file exists and is readable
+		if !helpers.FileIsReadable(file.unencrypted) {
+			return fmt.Errorf("cannot read input file %s", file.unencrypted)
+		}
+
+		// check that the output file doesn't exist
+		if helpers.FileExists(file.encrypted) {
+			return fmt.Errorf("outfile %s already exists", file.encrypted)
+		}
+	}
+
+	return nil
+}
+
+// Calculates md5 and sha256 hashes for the unencrypted and encrypted files
+func calculateHashes(fileSet encryptionFileSet) (*hashSet, error) {
+
+	hashes := hashSet{"", "", "", ""}
+
+	// open infile
+	unencryptedFile, err := os.Open(fileSet.unencrypted)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := unencryptedFile.Close(); err != nil {
+			log.Errorf("Error closing file: %s\n", err)
+		}
+	}()
+
+	// unencrypted md5 and sha256 checksums
+	md5Hash := md5.New()
+	shaHash := sha256.New()
+
+	tee := io.TeeReader(unencryptedFile, md5Hash)
+
+	_, err = io.Copy(shaHash, tee)
+	if err != nil {
+		return nil, err
+	}
+	hashes.unencryptedMd5 = hex.EncodeToString(md5Hash.Sum(nil))
+	hashes.unencryptedSha256 = hex.EncodeToString(shaHash.Sum(nil))
+
+	// encrypted md5 and sha256 checksums
+	encryptedFile, err := os.Open(fileSet.encrypted)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := encryptedFile.Close(); err != nil {
+			log.Errorf("Error closing file: %s\n", err)
+		}
+	}()
+
+	// encrypted md5
+	md5Hash.Reset()
+	shaHash.Reset()
+
+	tee = io.TeeReader(encryptedFile, md5Hash)
+	_, err = io.Copy(shaHash, tee)
+	if err != nil {
+		return nil, err
+	}
+	hashes.encryptedMd5 = hex.EncodeToString(md5Hash.Sum(nil))
+	hashes.encryptedSha256 = hex.EncodeToString(shaHash.Sum(nil))
+
+	return &hashes, nil
+}
+
+// Reads a public key file from a file using the crypt4gh keys module
+func readPublicKey(filename string) (key *[32]byte, err error) {
+	log.Info("Reading Public key file")
+	file, err := os.Open(filepath.Clean(filename))
+	if err != nil {
+		return nil, err
+	}
+
+	// This function panics if the key is malformed, so we handle that as well
+	// as errors
+	defer func() {
+		log.Info("Hey! this is a panic!")
```

Comment:
```suggestion
```
This message prints anyways. Either remove or move it inside the if statement.

Reply:
yes :D I just found it myself! I'll remove it ^^